shared/src/test/diff/nu/OptionalArgs.mls

```diff
@@ -0,0 +1,304 @@
+:NewDefs
+
+fun f1(a?: Int, b?: Int) = a + b
+//│ ╔══[ERROR] Type mismatch in operator application:
+//│ ║  l.3: 	fun f1(a?: Int, b?: Int) = a + b
+//│ ║       	                           ^^^^^
+//│ ╟── type `()` is not an instance of type `Int`
+//│ ║  l.3: 	fun f1(a?: Int, b?: Int) = a + b
+//│ ║       	           ^^^
+//│ ╟── but it flows into reference with expected type `Int`
+//│ ║  l.3: 	fun f1(a?: Int, b?: Int) = a + b
+//│ ╙──     	                           ^
+//│ TEST CASE FAILURE: There was an unexpected type error
+//│ fun f1: (a: Int | (), b: Int | ()) -> (Int | error)
+
+
+fun f1(a: Int, b: Int) = a + b
+//│ ╔══[ERROR] Type mismatch in operator application:
+//│ ║  l.17: 	fun f1(a: Int, b: Int) = a + b
+//│ ║        	                         ^^^^^
+//│ ╟── type `()` is not an instance of type `Int`
+//│ ║  l.17: 	fun f1(a: Int, b: Int) = a + b
+//│ ║        	          ^^^
+//│ ╟── but it flows into reference with expected type `Int`
+//│ ║  l.17: 	fun f1(a: Int, b: Int) = a + b
+//│ ╙──      	                         ^
```

Comment:
Hi @LPTK , I am having trouble with function parameter type parsing (seems all of the parameter types are '()' by default)

Reply:
NewParser.scala "?" parser -> [line](https://github.com/hkust-taco/mlscript/blob/7648ae331477f5c231c5c798b26d30bae68b5346/shared/src/main/scala/mlscript/NewParser.scala#L1428)

Reply:
That code does not make much sense to me. You may have mismerged the conflicts. The code is computing `(argName, argOpt)` by consuming an identifier, `?`, and `:` tokens. And then it redundantly checks optionality with `isOptional` – the logic is all messed up. I'm guessing the `()` might come from consuming the type tokens and then parsing an empty expression (which produces `undefined`, aka `()`), or it could be the later logic to make things optional that's broken. It might be better to start the parsing change from scratch.